apps/passport-server/src/services/e2eeService.ts

```diff
@@ -57,6 +60,62 @@ export class E2EEService {
 
     res.json(undefined satisfies UploadEncryptedStorageResponseValue);
   }
+
+  public async handleChangeBlobKey(
+    request: ChangeBlobKeyRequest,
+    res: Response
+  ): Promise<void> {
+    logger(
+      `[E2EE] Updating ${request.oldBlobKey} to ${request.newBlobKey} for ${request.uuid}`
+    );
+
+    if (
+      !request.newBlobKey ||
+      !request.oldBlobKey ||
+      !request.newSalt ||
+      !request.uuid ||
+      !request.encryptedBlob
+    ) {
+      throw new Error("Missing request fields");
+    }
+
+    // Ensure that old blob key is correct by checking if the row exists
+    const oldRow = await fetchEncryptedStorage(
+      this.context.dbPool,
+      request.oldBlobKey
+    );
+    if (!oldRow) {
+      res
+        .status(401)
+        .json({ error: { name: "PasswordInvalid" }, success: false });
+      return;
+    }
+
+    // Ensure that new salt is different from old salt
+    const commitment = await fetchCommitmentByUuid(
+      this.context.dbPool,
+      request.uuid
+    );
+    if (!commitment) {
+      throw new Error(`User not found with UUID ${request.uuid}`);
```

Comment:
Shouldn't this be a status of some sort?  404 or 400?

Reply:
This would throw a default 500 error right now. This wouldn't have any direct result on functionality, but I'll can add a custom error type (similar to "PasswordIncorrect") so that this error would be more clear to the client.

Reply:
Categorizing errors correctly between client and server is good discipline in general for properly reporting the health of your server.  It might be more sophisticated than we need to get in the near term, though, but not a bad habit to establish early.

My sensitivity to it might also be more relevant to cases where the client and server aren't both controlled by the same dev team.  When I used to get paged at 4am based on a server error rate, it was pretty important to me that somebody else's misbehaving code not wake me up. :)

Reply:
Totally makes sense - error visibility is one of those low-cost, high-impact additions that can make a huge difference when we inevitability get paged on rollbar